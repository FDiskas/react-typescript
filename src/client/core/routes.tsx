import * as React from 'react';
import { Switch } from 'react-router-dom';
import { Route } from 'react-router4-with-layouts';

import { HomePage } from '../pages/home/HomePage';
import { DemoPage } from '../pages/demo/DemoPage';
import { LoginPage } from '../pages/login/LoginPage';
import { DefaultLayout } from '../layouts/default/DefaultLayout';
import { EmptyLayout } from '../layouts/empty/EmptyLayout';

export const CoreRoutes = () => (
    <Switch>
        <Route path="/" component={HomePage} exact={true} layout={DefaultLayout} />
        <Route path="/demo" component={DemoPage} layout={DefaultLayout} />
        <Route path="/login" component={LoginPage} layout={EmptyLayout} />
    </Switch>
);
