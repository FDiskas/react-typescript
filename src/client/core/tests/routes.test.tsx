import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { CoreRoutes } from '../routes';

describe('routes.js', () => {
    it('renders without crashing', () => {
        shallow(<CoreRoutes/>);
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<CoreRoutes/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
