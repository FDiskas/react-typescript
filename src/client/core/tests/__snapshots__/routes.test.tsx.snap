// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes.js matches snapshot 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    layout={[Function]}
    path="/"
  />
  <Route
    component={[Function]}
    layout={[Function]}
    path="/demo"
  />
  <Route
    component={[Function]}
    layout={[Function]}
    path="/login"
  />
</Switch>
`;
